superset-frontend/src/dashboard/components/DashboardBuilder/state.ts

```diff
@@ -29,6 +29,9 @@ import {
 // eslint-disable-next-line import/prefer-default-export
 export const useNativeFilters = () => {
   const [isInitialized, setIsInitialized] = useState(false);
+  const showNativeFilters = useSelector<RootState, boolean>(
+    state => getUrlParam(URL_PARAMS.showFilters) ?? true,
```

Comment:
@Vitor-Avila The [original code](https://github.com/apache/superset/pull/23228/files#diff-6f2c6562ff1fad0ce405b3507cddff63a25dc91c76fb8613284b9cbb241dd928) had `&& state.dashboardInfo.metadata?.show_native_filters`. Isn't that needed anymore?

Reply:
@michael-s-molina my understanding is that the original PR fully removed `metadata.show_native_filters` (from all places). I suspect the reason for it is that this metadata config was intended to permanently hide the native filters on dashboards that had filter box, until it was fully migrated. So I don't think we need that back. If we want to bring that logic, we would need to re-update the schemas, and logic in other places too.

Not sure if you had a chance to test in that ephemeral but I believe things are working properly with these changes.